1.0/docs/devguide/registering-elements.md

```diff
@@ -177,17 +177,75 @@ all imports in the document have finished loading.
 ## Lifecycle callbacks {#lifecycle-callbacks}
 
 Polymer's Base prototype implements the standard Custom Element lifecycle
-callbacks to perform tasks necessary for Polymer's built-in features.  The hooks
-in turn call shorter-named lifecycle methods on your prototype.
-
-- `created` instead of `createdCallback`
-- `attached` instead of `attachedCallback`
-- `detached` instead of `detachedCallback`
-- `attributeChanged` instead of `attributeChangedCallback`
+callbacks to perform tasks necessary for Polymer's built-in features.
+{{site.project_title}} in turn calls shorter-named lifecycle methods on your
+prototype.
 
 Polymer adds an extra callback, `ready`, which is invoked when Polymer has
 finished creating and initializing the element's local DOM.
 
+<table>
+  <tr>
+    <th>Callback</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td><code>created</code></td>
+    <td>Called when the element has been created, but before property values are
+       set and local DOM is initialized.
+      <p>Use for one-time set-up before property values are set.
+      </p>
+      <p>Use instead of <code>createdCallback</code>.
+      </p>
+    </td>
+  </tr>
+  <tr>
+    <td><code>ready</code></td>
+    <td>Called after property values are set and local DOM is initialized.
+      <p>Use for one-time configuration of your component after local
+        DOM is initialized. (For configuration based on property values, it
+        may be preferable to use an <a href="properties.html#multi-property-observers">observer</a>.)
+      </p>
+    </td>
+  </tr>
+  <tr>
+    <td><code>attached</code></td>
+    <td>Called after the element is attached to the document.
+      <p>Uses include accessing computed style information, and adding
+        document-level event listeners. (If you use declarative
+        event handling, such as <a href="events.html#annotated-listeners">annotated
+        event listeners</a> or the
+        <a href="events.html#event-listeners"><code>listeners</code> object</a>,
+        {{site.project_title}} automatically adds listeners on attach and removes
+        them on detach.)
+      </p>
+      <p>Use instead of <code>attachedCallback</code>.
+      </p>
+    </td>
+  </tr>
+  <tr>
+    <td><code>detached</code></td>
+    <td>Called after the element is attached to the document.
```

Comment:
It might be useful to reword these so it's more clear that attached and detached aren't one-off callbacks, but happen every time the element is attached / detached. I think the use of 'Called *after*' above in the explanation of ready kind of primed me to read it that way. 'Called *when*', like in attributeChanged, seems to imply that this can occur multiple times, but it also loses the important detail that it occurs after attachment / detachment. Maybe there's another way to state it.